src/ink.tsx

```diff
@@ -0,0 +1,184 @@
+import React, {ReactNode} from 'react';
+import throttle from 'lodash.throttle';
+import logUpdate, {LogUpdate} from 'log-update';
+import isCI from 'is-ci';
+import {reconciler} from './reconciler';
+import {createRenderer, InkRenderer} from './renderer';
+import signalExit from 'signal-exit';
+import {createNode, DOMElement} from './dom';
+import {FiberRoot} from 'react-reconciler';
+import {instances} from './instances';
+import {App} from './components/App';
+
+export interface InkOptions {
+	stdout: NodeJS.WriteStream;
+	stdin: NodeJS.ReadStream;
+	debug: boolean;
+	exitOnCtrlC: boolean;
+	experimental: boolean;
+	waitUntilExit?: () => Promise<void>;
+}
+
+export interface Ink {
```

Comment:
Only these props should be publicly available - https://github.com/vadimdemedes/ink#instance. I commented above on having an old `Instance` class, let's discuss there.